### PR TITLE
Fix eslint issues in index.js

### PR DESCRIPTION
**Pull request by c3ho**
## Issue This PR Addresses
Part of #990 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Refactored our async functions existing outside of `useEffect()` hooks to use `useCallback()`. The only es lint warnings remaining are on line 84 and from my testing, adding the dependencies `post` or `nextPageLink` as recommended will cause the issue in #1018 

I kind of hate the front-end and its dependencies...
From what I understand:
1. `useEffect` line 108 is called every re-render

2. `useEffect` on line 113 is caused by either `getPostsCount`(async function) or change to `currentNumPosts`(state). This sets the current and initial # of posts to whatever the current # of posts in redis is

3. `useEffect` on line 127 should be caused by the callback on line 108? This causes it to start the timer to get the updated # of posts every 5 mins and set currentNumPosts to the new #. Once this is set, it should cause the `useEffect` from 2.

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by c3ho** [comment]
I kind of hate the front-end and its dependencies...

From what I understand:
1. `useEffect` line 108 is called every re-render

2. `useEffect` on line 113 is triggered by either `getPostsCount`(async function) or change to `currentNumPosts`(state). This sets the current and initial # of posts to whatever the current # of posts in redis is

3. `useEffect` on line 127 should be triggered by the callback on line 108? This causes it to start the timer to get the updated # of posts every 5 mins and set currentNumPosts to the new #. Once this is set, it should trigger the `useEffect` from 2.

File: src/frontend/src/pages/index.js [hunk 1]
```diff
@@ -124,18 +122,17 @@ export default function IndexPage() {
       }
     }
     setPostsInfo();
-  }, []);
+  }, [getPostsCount, currentNumPosts]);
```
Comment (humphd) [inline src/frontend/src/pages/index.js:125]: Does this not need `initNumPosts` too?  Also, why do we call `getPostsCount()` twice on 117 and 118, and not simply re-use the value?

Comment (c3ho) [reply src/frontend/src/pages/index.js:125]: We don't care about whatever initNumPosts is going to be in this case. Both currentNumPosts and initNumPosts will be overwritten. I only want this side-effect happening whenever a currentNumPosts value changes

File: src/frontend/src/pages/index.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ export default function IndexPage() {
     setNumPages(numPages + 1);
   }
 
-  async function getPostsCount() {
+  const getPostsCount = useCallback(async () => {
```
Comment (humphd) [inline src/frontend/src/pages/index.js:91]: I'm unclear if you can use an `async` function here.  What I read online and in the docs doesn't help much.

Comment (c3ho) [reply src/frontend/src/pages/index.js:91]: https://stackoverflow.com/questions/60835660/proper-use-of-usecallback . Looks like this guy does it in the React hook

**Comment by c3ho** [comment]
Will merge after rebase as it received two approvals

**Pull request merged**